light-client/src/client_state.rs

```diff
@@ -157,6 +157,12 @@ impl TryFrom<RawClientState> for ClientState {
             .as_ref()
             .ok_or(Error::MissingLatestHeight)?;
 
+        if raw_latest_height.revision_number != 0 {
```

Comment:
While this modifies the same lines of code as BELC3 #66 , since the BELC3 fixes also resolve the BELC4 problems, this PR doesn't appear to be necessary for merging.

https://github.com/datachainlab/parlia-elc/pull/66/files#diff-e8516aac7f0e5a1053a37ce52b1d1a7829f38b993b21df81a8c4726e392038a0R162